engine/runtime/src/test/java/org/enso/interpreter/test/HashCodeTest.java

```diff
@@ -0,0 +1,145 @@
+        .option(
+            RuntimeOptions.LANGUAGE_HOME_OVERRIDE,
+            Paths.get("../../distribution/component").toFile().getAbsolutePath()
+        ).build();
+    final Map<String, Language> langs = ctx.getEngine().getLanguages();
+    assertNotNull("Enso found: " + langs, langs.get("enso"));
+    var src = Source.newBuilder("enso", """
+        equals x y = x == y
+        hashCode x = x.hash_code
+        """, "tmp.enso").buildLiteral();
+    var module = ctx.eval(src);
+    equalsFunc = module.invokeMember(Module.EVAL_EXPRESSION, "equals");
+    hashCodeFunc = module.invokeMember(Module.EVAL_EXPRESSION, "hashCode");
+    assertTrue(equalsFunc.canExecute());
+    assertTrue(hashCodeFunc.canExecute());
+    this.valGenerator = ValuesGenerator.create(
+        ctx,
+        ValuesGenerator.Language.ENSO,
+        ValuesGenerator.Language.JAVA,
+        ValuesGenerator.Language.JAVASCRIPT,
+        ValuesGenerator.Language.PYTHON
+    );
+  }
+
+  @After
+  public void closeContext() {
+    this.ctx.close();
+  }
+
+  @Test
+  public void testPrimitives() {
+    for (var primitiveValues : Arrays.asList(
+        valGenerator.booleans(),
+        valGenerator.numbers(),
+        valGenerator.textual()
+    )) {
+      checkHashContract(primitiveValues);
+    }
+  }
+  @Test
+  public void testVectors() {
+    checkHashContract(valGenerator.vectors());
+  }
+
+  @Test
+  public void testArrayLikeValues() {
+    checkHashContract(valGenerator.arrayLike());
+    checkHashContract(valGenerator.vectors());
+  }
+
+  @Test
+  public void testTimesAndDates() {
+    for (var dateTimeValues : Arrays.asList(
+        valGenerator.timesAndDates(),
+        valGenerator.timeZones(),
+        valGenerator.durations(),
+        valGenerator.periods()
+    )) {
+      checkHashContract(dateTimeValues);
+    }
+  }
+
+  private void checkHashContract(List<Value> values) {
+    checkHashContract(values, values);
+  }
+
+  /**
+   * All the elements of firstVals and secondVals should have same type.
+   * <p>
+   * See the description of hashing contract
+   * in {@link org.enso.interpreter.node.expression.builtin.meta.HashCodeAnyNode}.
+   */
+  private void checkHashContract(List<Value> firstVals, List<Value> secondVals) {
+    for (Value firstVal : firstVals) {
+      for (Value secondVal : secondVals) {
+        if (valuesEqual(firstVal, secondVal)) {
+          long firstHash = valueHash(firstVal);
+          long secondHash = valueHash(secondVal);
+          assertEquals(
+              String.format("""
+              If two objects are same, they should have same hash codes:
+                firstVal = %s, secondVal = %s, firstHash = %d, secondHash = %d
+              """, firstVal, secondVal, firstHash, secondHash),
+              firstHash,
+              secondHash
+          );
+        }
+      }
+    }
+  }
+
+  private long valueHash(Value value) {
+    Value hash = hashCodeFunc.execute(value);
```

Comment:
I think following shall be good enough for a test:
```
var n = HashCodeAnyNode.create();
var f = value.getClass().getSuperclass().getDeclaredField("reciever");
f.setAccessible(true);
var v = f.get(value);
var hash = n.execute(v);
```
I hope it works.

Reply:
As you suggested in our call, I successfully use simple executable TruffleObject for unwrapping values. Your code fails in `f.setAccessible(true)` on InaccessibleObjectException "module org.graalvm.sdk does not open org.graalvm.polyglot to unnamed module"